components/MapsDedi/index.vue

```diff
@@ -41,15 +41,97 @@
           </div>
         </div>
       </div>
+      <div v-show="isDesa" class="maps">
+        <div class="maps__motif">
+          <img src="~/assets/images/MotifDot.svg" alt="Motif Dot">
+        </div>
+        <div
+          ref="boxmaps"
+          :class="{
+            'maps__boxmaps--fullscreen': isFullscreen,
+            'maps__boxmaps--exitfullscreen': !isFullscreen,
+            'maps__boxmaps': true,
+
+          }"
+        >
+          <div
+            ref="maps"
+            :class="{
+              'maps__boxmaps-map--fullscreen': isFullscreen,
+              'maps__boxmaps-map--exitfullscreen': !isFullscreen,
+              'maps__boxmaps-map': true
+
+            }"
+          />
+          <div
+            :class="{
+              'maps__boxmaps-info':true,
+              'maps__boxmaps-info--fullscreen': isFullscreen,
+              'maps__boxmaps-info--exitfullscreen': !isFullscreen,
+            }"
+          >
+            <div class="maps__boxmaps-info-box">
+              <div class="maps__boxmaps-info-box-title">
+                Total <span class="font-bold">{{ totalVillage }} Desa</span> telah bergabung
+              </div>
+              <div class="maps__boxmaps-info-box-subtitle">
+                Update per {{ new Date().toLocaleString('id', { month: 'long' }) }} {{ new Date().getFullYear() }}
+              </div>
+            </div>
+          </div>
+          <div
+            :class="{
+              'maps__boxmaps-control':true,
+              'maps__boxmaps-control--fullscreen': isFullscreen,
+              'maps__boxmaps-control--exitfullscreen': !isFullscreen,
+            }"
+          >
+            <div ref="currenlocation" class="maps__boxmaps-control-current-location" @click="setCurrentLocation">
+              <img src="~/assets/icons/IconCurrentLocation.svg" class="w-5 h-5" alt="Icon Current Location">
+            </div>
+            <div>
+              <div class="maps__boxmaps-control-zoom-in" @click="setZoomIn()">
+                <img src="~/assets/icons/IconPlus.svg" class="w-5 h-5" alt="Icon Plus">
+              </div>
+              <div class="maps__boxmaps-control-zoom-out" @click="setZoomOut()">
+                <img src="~/assets/icons/IconMinus.svg" class="w-5 h-5" alt="Icon Plus">
+              </div>
+            </div>
+            <div ref="fullscreen" class="maps__boxmaps-control-fullscreen" @click="setFullscreen">
+              <img src="~/assets/icons/IconFullscreen.svg" class="w-5 h-5" alt="Icon Fullscreen">
+            </div>
+          </div>
+        </div>
+      </div>
     </div>
   </div>
 </template>
 
 <script>
+import { Loader } from '@googlemaps/js-api-loader'
+import { MarkerClusterer, KmeansAlgorithm } from '@googlemaps/markerclusterer'
+
+const loader = new Loader({
+  apiKey: 'AIzaSyBisT1SjBCXqWjGdUT2Iv7LnbdCWKU__7w',
```

Comment:
is this API Key considered as a credential? should we put them on env variables? 🤔

Reply:
i thinks that not problem, you can visit all web using google maps, and inspect the html and then you can find the apiKey of google maps. google maps api key must register the domain where we install the application so that not just anyone can use it